jbpm-services/jbpm-kie-services/src/main/java/org/jbpm/kie/services/impl/bpmn2/BPMN2DataServiceImpl.java

```diff
@@ -47,17 +60,32 @@
 public class BPMN2DataServiceImpl implements DefinitionService, DeploymentEventListener {
     
     private static final Logger logger = LoggerFactory.getLogger(BPMN2DataServiceImpl.class);
+    
     private static final BPMN2DataServiceSemanticModule MODULE = new BPMN2DataServiceSemanticModule();
+    private static final BPMN2DataServiceExtensionSemanticModule EXTENSIONS_MODULE = new BPMN2DataServiceExtensionSemanticModule(MODULE);
     
-    private ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
+    private ConcurrentMap<String, Map<String, ProcessDescRepoHelper>> definitionCache = 
     		new ConcurrentHashMap<String, Map<String, ProcessDescRepoHelper>>();
+   
+    private static final String [] dialectNames = { "java", "mvel" };
+    private static final ProcessDialect [] dataServiceDialects = { 
+            new DataServiceJavaProcessDialect(),
+            new DataServiceMvelProcessDialect()
```

Comment:
these are just extensions to the default ones, meaning they provide same capabilities so it's safe to be replaced as it affects overall parsing of the processes as it's static stuff?

Was thinking if we can use just one implementation to avoid this extensions in services. Though it might be too big change right now... ideally it would be good to have this "analysis" done on core engine level and provide that to the services somehow - kbase model repository or so... just an idea for future